### PR TITLE
Add test for @include over @internal

### PR DESCRIPTION
**Pull request by tadelesh**
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

This test is from openai real use case: https://github.com/Azure/autorest.csharp/issues/3519. The new added test could not be tested by mock server, different languages' SDK should add test for their generated code.
The logic emitter should accomplish:
1. `isInternal` will return `true` for `InternaIncludelModel` model
2. `isInclude` will return `true` for `InternaIncludelModel` model
3. `InternaIncludelModel` will be included in `getAllModels`
4. SDK should generate `InternaIncludelModel` and make it public to user.

### CODE REVIEW

File: .changeset/nine-otters-change.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@azure-tools/cadl-ranch-specs": patch
+---
+
+Add test for @include over @internal
```
Comment (timotheeguerin) [inline .changeset/nine-otters-change.md:5]:
```suggestion
Add test for `@include` over `@internal`
```
nit: so it doesn't tag people

File: packages/cadl-ranch-specs/http/azure/client-generator-core/internal/main.tsp [hunk 2]
```diff
@@ -20,6 +20,12 @@ model InternalModel {
   name: string;
 }
 
+@doc("This is a model only used by internal operation. Also, it is decorated with @include. It should be generated and exported.")
+@global.Azure.ClientGenerator.Core.include
+model InternaIncludelModel {
```
Comment (timotheeguerin) [inline packages/cadl-ranch-specs/http/azure/client-generator-core/internal/main.tsp:25]:
```suggestion
model InternalIncludeModel {
```

typo

File: packages/cadl-ranch-specs/http/azure/client-generator-core/internal/main.tsp [hunk 3]
```diff
@@ -20,6 +20,12 @@ model InternalModel {
...
+  name: string;
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/client-generator-core/internal/main.tsp:26]: Should we add a nested model to also test transitivity?

Or, if we don't sure how transitivity works for `@include`, I am find to leave as this.

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

Please take care of Timothee's comment.

PS: whether model is output-only or input-output actually affect Java class (guess .NET too), but that's language problem, no need to worry in tsp test.

File: packages/cadl-ranch-specs/http/azure/client-generator-core/internal/main.tsp (hunk 3 above)
Comment (tadelesh) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/internal/main.tsp:26]: I think transitivity should be the work of `isInclude` (though it seems current TCGC has not implemented it yet). I've created an [issue](https://github.com/Azure/typespec-azure/issues/3264) to track..

**Comment by tadelesh** [comment]
@msyyc @qiaozha @pshao25 please review from your language's perspective.

File: packages/cadl-ranch-specs/http/azure/client-generator-core/internal/main.tsp (hunk 3 above)
Comment (pshao25) [reply packages/cadl-ranch-specs/http/azure/client-generator-core/internal/main.tsp:26]: I think we should figure out this issue now.

**Review by pshao25 — CHANGES_REQUESTED** [review]
Checked the user's spec, it has nested model. So let's figure out how to handle it in this PR.

**Comment by tadelesh** [comment]
> Checked the user's spec, it has nested model. So let's figure out how to handle it in this PR.

Added nested model in current test, and a fix for @include transitivity: https://github.com/Azure/typespec-azure/pull/3267.

**Pull request closed without merging**